### PR TITLE
Add deprecation notice to manager methods

### PR DESCRIPTION
**Pull request by maarcingebala**
Port https://github.com/saleor/saleor/pull/16587 to main. For now deprecations are moved to main for consistency with 3.20. Webhooks will be moved to the core in smaller chunks on the main from now on.

# Impact

- [ ] New migrations
- [ ] New/Updated API fields or mutations
- [ ] Deprecated API fields or mutations
- [ ] Removed API types, fields, or mutations

# Docs

- [ ] Link to documentation:

# Pull Request Checklist

- [ ] Privileged queries and mutations are either absent or guarded by proper permission checks
- [ ] Database queries are optimized and the number of queries is constant
- [ ] Database migrations are either absent or optimized for zero downtime
- [ ] The changes are covered by test cases
- [ ] All new fields/inputs/mutations have proper labels added (`ADDED_IN_X`, `PREVIEW_FEATURE`, etc.)
- [ ] All migrations have proper dependencies
- [ ] All indexes are added concurrently in migrations
- [ ] All RunSql and RunPython migrations have revert option defined

### CODE REVIEW

**Review by korycins — APPROVED** [review]

File: saleor/plugins/webhook/plugin.py [hunk 1]
```diff
@@ -3212,6 +3212,8 @@ def get_taxes_for_order(
                 self.requestor,
             )
 
+    # Note: this method is deprecated in Saleor 3.20 and will be removed in Saleor 3.21.
```
Comment (korycins) [inline saleor/plugins/webhook/plugin.py:3215]: Why did you add this warning for this handler but didn't add it to the rest of WebhookPlugin handlers? 🤔 This can be misleading as WebhookPlugin.get_shipping_methods_for_checkout( will be just moved to core, but it will work in the same way as previously.

**Review by kadewu — DISMISSED** [review]

File: saleor/plugins/webhook/plugin.py (hunk 1 above)
Comment (maarcingebala) [reply saleor/plugins/webhook/plugin.py:3215]: Yes, I think that's a mistake; I'll revert this. Deprecations are added in manager and base plugin.

**Review by fowczarek — APPROVED** [review]

**Review by IKarbowiak — APPROVED** [review]
